puppeth: YOLOv1 nil pointer dereference

puppeth panics when configuring new genesis.

#### System information

Geth version: `geth version`
OS & Version: Linux
Commit hash : a3b5c762376c8ceec388d238d293c9ef76bd6549

#### Expected behaviour

no panic

#### Actual behaviour

panic

#### Steps to reproduce the behaviour

create new genesis with puppeth.
configure new genesis.
answer questions until YOLOv1 block.

#### Backtrace

````
Which block should YOLOv1 (EIP2537) come into effect? (default = <nil>)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x869fe4]

goroutine 1 [running]:
main.(*wizard).manageGenesis(0xc000150100)
        github.com/ethereum/go-ethereum/cmd/puppeth/wizard_genesis.go:244 +0x1804
main.(*wizard).run(0xc000150100)
        github.com/ethereum/go-ethereum/cmd/puppeth/wizard_intro.go:149 +0xa08
main.runWizard(0xc0001269a0, 0x3, 0xf69fe0)
        github.com/ethereum/go-ethereum/cmd/puppeth/puppeth.go:63 +0x27a
gopkg.in/urfave/cli%2ev1.HandleAction(0x8de200, 0x9b94c0, 0xc0001269a0, 0xc000078780, 0x0)
        gopkg.in/urfave/cli.v1@v1.20.0/app.go:490 +0xc8
gopkg.in/urfave/cli%2ev1.(*App).Run(0xc0001096c0, 0xc0000201e0, 0x1, 0x1, 0x0, 0x0)
        gopkg.in/urfave/cli.v1@v1.20.0/app.go:264 +0x5dd
main.main()
        github.com/ethereum/go-ethereum/cmd/puppeth/puppeth.go:54 +0x24b
````

--- a/cmd/puppeth/wizard_genesis.go
+++ b/cmd/puppeth/wizard_genesis.go
@@ -30,6 +30,7 @@ import (
 	"time"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/params/confp/tconvert"
 	"github.com/ethereum/go-ethereum/params/types/ctypes"
@@ -201,47 +202,57 @@ func (w *wizard) manageGenesis() {
 	fmt.Println(" 2. Export genesis configurations")
 	fmt.Println(" 3. Remove genesis configuration")
 
+	// sanitizeUint64P safeguards against taking the value of a potential nil address.
+	// In case a nil is passed, the max integer value is returned, which in the chain configuration context
+	// is functionally equivalent to being unset.
+	sanitizeUint64P := func(n *uint64) uint64 {
+		if n == nil {
+			return math.MaxUint64
+		}
+		return *n
+	}
+
 	choice := w.read()
 	switch choice {
 	case "1":
 		// Fork rule updating requested, iterate over each fork
 		fmt.Println()
-		fmt.Printf("Which block should Homestead come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEthashHomesteadTransition())
-		w.conf.Genesis.Config.SetEthashHomesteadTransition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEthashHomesteadTransition()))
+		fmt.Printf("Which block should Homestead come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEthashHomesteadTransition()))
+		w.conf.Genesis.Config.SetEthashHomesteadTransition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEthashHomesteadTransition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should EIP150 (Tangerine Whistle) come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP150Transition())
-		w.conf.Genesis.Config.SetEIP150Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP150Transition()))
+		fmt.Printf("Which block should EIP150 (Tangerine Whistle) come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP150Transition()))
+		w.conf.Genesis.Config.SetEIP150Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP150Transition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should EIP155 (Spurious Dragon) come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP155Transition())
-		w.conf.Genesis.Config.SetEIP155Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP155Transition()))
+		fmt.Printf("Which block should EIP155 (Spurious Dragon) come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP155Transition()))
+		w.conf.Genesis.Config.SetEIP155Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP155Transition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should EIP158/161 (also Spurious Dragon) come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP161dTransition())
-		w.conf.Genesis.Config.SetEIP161dTransition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP161dTransition()))
+		fmt.Printf("Which block should EIP158/161 (also Spurious Dragon) come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP161dTransition()))
+		w.conf.Genesis.Config.SetEIP161dTransition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP161dTransition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should Byzantium come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEthashEIP649Transition())
-		w.conf.Genesis.Config.SetEthashEIP649Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEthashEIP649Transition()))
+		fmt.Printf("Which block should Byzantium come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEthashEIP649Transition()))
+		w.conf.Genesis.Config.SetEthashEIP649Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEthashEIP649Transition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should Constantinople come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEthashEIP1234Transition())
-		w.conf.Genesis.Config.SetEthashEIP1234Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEthashEIP1234Transition()))
+		fmt.Printf("Which block should Constantinople come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEthashEIP1234Transition()))
+		w.conf.Genesis.Config.SetEthashEIP1234Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEthashEIP1234Transition())))
 		if w.conf.Genesis.Config.GetEIP1283DisableTransition() == nil {
 			w.conf.Genesis.Config.SetEIP1283DisableTransition(w.conf.Genesis.Config.GetEthashEIP1234Transition())
 		}
 		fmt.Println()
-		fmt.Printf("Which block should Petersburg come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP1283DisableTransition())
-		w.conf.Genesis.Config.SetEIP1283DisableTransition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP1283DisableTransition()))
+		fmt.Printf("Which block should Petersburg come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP1283DisableTransition()))
+		w.conf.Genesis.Config.SetEIP1283DisableTransition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP1283DisableTransition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should Istanbul come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP145Transition())
-		w.conf.Genesis.Config.SetEIP145Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP145Transition()))
+		fmt.Printf("Which block should Istanbul come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP145Transition()))
+		w.conf.Genesis.Config.SetEIP145Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP145Transition())))
 
 		fmt.Println()
-		fmt.Printf("Which block should YOLOv1 (EIP2537) come into effect? (default = %v)\n", w.conf.Genesis.Config.GetEIP2537Transition())
-		w.conf.Genesis.Config.SetEIP2537Transition(w.readDefaultUint64P(*w.conf.Genesis.Config.GetEIP2537Transition()))
+		fmt.Printf("Which block should YOLOv1 (EIP2537) come into effect? (default = %v)\n", sanitizeUint64P(w.conf.Genesis.Config.GetEIP2537Transition()))
+		w.conf.Genesis.Config.SetEIP2537Transition(w.readDefaultUint64P(sanitizeUint64P(w.conf.Genesis.Config.GetEIP2537Transition())))
 
 		out, _ := json.MarshalIndent(w.conf.Genesis.Config, "", "  ")
 		fmt.Printf("Chain configuration updated:\n\n%s\n", out)

